### PR TITLE
CI: add GHA-based CI

### PR DESCRIPTION
**Pull request by thesamesam**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install tox
+        run: pip install tox
+      - name: Test using tox
+        run: tox -e py
+
```
Comment (mgorny) [inline .github/workflows/ci.yaml:21]: Ohnoes, a trailing empty line!

**Comment by mgorny** [comment]
Any clue how to test it? I think gentoopm's test suite doesn't work outside Gentoo because of PMs being, err.

**Comment by thesamesam** [comment]
> Any clue how to test it? I think gentoopm's test suite doesn't work outside Gentoo because of PMs being, err.

I was trying to figure that out but on my fork, it hangs forever before even running `tox`!

See https://github.com/thesamesam/gentoopm/actions/runs/4465327586/jobs/7842287948. I was going to see about a Gentoo image but there's no point if I can't make it even run 😞

**Comment by thesamesam** [comment]
ha, I didn't realise tox doesn't actually work right now. Oh well.

**Pull request closed without merging**